carl/envs/rna/carl_rna.py

```diff
@@ -0,0 +1,119 @@
+        gaussian_noise_std_percentage: float = 0.01,
+        logger: Optional[TrialLogger] = None,
+        scale_context_features: str = "no",
+        default_context: Optional[Context] = DEFAULT_CONTEXT,
+        max_episode_length: int = 500,
+        state_context_features: Optional[List[str]] = None,
+        context_mask: Optional[List[str]] = None,
+        dict_observation_space: bool = False,
+        context_selector: Optional[
+            Union[AbstractSelector, type[AbstractSelector]]
+        ] = None,
+        context_selector_kwargs: Optional[Dict] = None,
+    ):
+        """
+        Parameters
+        ----------
+        env: gym.Env, optional
+            Defaults to classic control environment mountain car from gym (MountainCarEnv).
+        contexts: List[Dict], optional
+            Different contexts / different environment parameter settings.
+        instance_mode: str, optional
+        """
+        if not contexts:
+            contexts = {0: DEFAULT_CONTEXT}
+        if env is None:
+            env_config = RnaDesignEnvironmentConfig(
+                mutation_threshold=DEFAULT_CONTEXT["mutation_threshold"],
+                reward_exponent=DEFAULT_CONTEXT["reward_exponent"],
+                state_radius=DEFAULT_CONTEXT["state_radius"],
+            )
+            dot_brackets = parse_dot_brackets(
+                dataset=DEFAULT_CONTEXT["dataset"],
+                data_dir=data_location,
+                target_structure_ids=DEFAULT_CONTEXT["target_structure_ids"],
+            )
+            env = RnaDesignEnvironment(dot_brackets, env_config)
+
+        env.action_space = ACTION_SPACE
+        env.observation_space = OBSERVATION_SPACE
+        env.reward_range = (-np.inf, np.inf)
+        env.metadata = {}
+        # The data_location in the RNA env refers to the place where the dataset is downloaded to, so it is not changed
+        # with the context.
+        env.data_location = data_location
+        super().__init__(
+            env=env,
+            contexts=contexts,
+            hide_context=hide_context,
+            add_gaussian_noise_to_context=add_gaussian_noise_to_context,
+            gaussian_noise_std_percentage=gaussian_noise_std_percentage,
+            logger=logger,
+            scale_context_features=scale_context_features,
+            default_context=default_context,
+            max_episode_length=max_episode_length,
+            state_context_features=state_context_features,
+            dict_observation_space=dict_observation_space,
+            context_selector=context_selector,
+            context_selector_kwargs=context_selector_kwargs,
+            context_mask=context_mask,
+        )
+        self.whitelist_gaussian_noise = list(DEFAULT_CONTEXT)
+
+    def step(self, action):
+        # Step function has a different name in this env
+        state, reward, done = self.env.execute(action)
+        if not self.hide_context:
```

Comment:
Why do we concat the state here? `CARLEnv` is supposed to handle this. Or is this a special case? Please comment or adjust

Reply:
Done